astropy/time/core.py

```diff
@@ -437,6 +438,9 @@ def precision(self, val):
             raise ValueError('precision attribute must be an int between '
                              '0 and 9')
         self._precision = val
+        if hasattr(self, '_time'):
```

Comment:
@mhvk - I've implemented your suggestion because of the caching advantage for `value`.

I think that it is still always necessary to clear the cache when `precision` or `out_subfmt` are modified.  (I did `in_subfmt` just for consistency).

Now about pushing the modified value of `precision` down into `_time`... I think that given the original design (mistake) of not supplying the parent `Time` object to the child `TimeFormat` object, we are stuck with this minor inelegance to maintain consistency.  However, there *is* an alternative of always calling replicate, so change from:
```
        elif attr in self.FORMATS:
            cache = self.cache['format']
            if attr not in cache:
                if attr == self.format:
                    tm = self
                else:
                    tm = self.replicate(format=attr)
                value = tm._shaped_like_input(tm._time.to_value(parent=tm))
                cache[attr] = value
```
to
```
        elif attr in self.FORMATS:
            cache = self.cache['format']
            if attr not in cache:
                tm = self.replicate(format=attr)
                value = tm._shaped_like_input(tm._time.to_value(parent=tm))
                cache[attr] = value
```

Option 1) is to keep `_time` in sync with the parent (for the value formatting attributes).
Option 2) is to allow `_time` to get out of sync with the parent (for the value formatting attributes), having the assurance that to actually access those values it will get synced up via the call to replicate.

My inclination would be option 1 since there is less opportunity for foot-shooting later if something changes.  But it does come at the price of an inelegant coupling with `_time`.